spark/dl/src/main/scala/com/intel/analytics/bigdl/parameters/ParameterManagerMasterEndpoint.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sparkExtension
+
+import java.util.HashMap
+
+import org.apache.spark.{SparkConf, SparkContext, SparkEnv}
+import org.apache.spark.rpc._
+import org.apache.spark.storage.{BlockId, BlockManagerId}
+import org.apache.spark.util.{RpcUtils, Utils}
+import org.apache.spark.SecurityManager
+
+import scala.collection.mutable
+
+private case class GetExecutorBlockList(executorId: Int)
+private case class UpdateExecutorBlockList(executorId: Int, blockId: BlockId)
+private case class ClearExecutorBlockList(executorId: Int)
+private case class ResetTotal()
+
+/**
+ * ParameterManagerMasterEndpoint is an [[ThreadSafeRpcEndpoint]] on the driver
+ * node to track statuses of all executor' blocks.
+ */
+private class ParameterManagerMasterEndpoint(
+  override val rpcEnv: RpcEnv,
+  conf: SparkConf)
+  extends ThreadSafeRpcEndpoint {
+
+  private val blocks = new HashMap[Int, mutable.HashSet[BlockId]]
```

Comment:
why not use a concurrent hashmap?

Reply:
Test results shows currently overhead of executor sync with master about block status is not noticeable, i think current logic should be fine.